index.html

```diff
@@ -117,21 +117,31 @@ <h2>API</h2>
 import * as parsel from "./parsel.js";
 self.parsel = parsel; // so that people can experiment in the console
 
+if(location.search.includes('?selector=')) {
+
+	let selector = decodeURIComponent(location.href.split('?selector=')[1])
+	selectorText.value = selector || selectorText.value // set default value if selector is empty
+	
+	history.replaceState(null, '', `?selector=${selector}`)
+}
+
 (tryout.oninput = e => {
-	tokens.textContent = JSON.stringify(parsel.tokenize(selectorText.value), null, "\t");
-	Prism.highlightElement(tokens);
-
-	try {
-
-		tree.textContent = JSON.stringify(parsel.parse(selectorText.value, {recursive: recursive.checked}), null, "\t");
-		tree.classList.remove("error");
-		specificity.textContent = parsel.specificity(selectorText.value);
-		Prism.highlightElement(tree);
-	}
-	catch (e) {
-		tree.classList.add("error");
-		tree.innerHTML = `<details><summary>${e}</summary>${e.stack.replace(e, "")}</details>`;
-	}
+history.replaceState(null, '', `?selector=${selectorText.value}`)
```

Comment:
This will overwrite any other params, as well as any hash. Better use `new URL(location)` and `url.searchParams` on that.

Reply:
url.searchParams returns an empty object since there are hashs in the url. How i can fix it?

Reply:
So what would happen if i keep the url first, then remove the hashes from the url, then use `new URL(location)` and `url.searchParams`.

What do you think?

Reply:
`url.searchParams` shouldn't return an empty object due to hashes. It doesn't deal with hashes at all. See https://developer.mozilla.org/en-US/docs/Web/API/URL/searchParams
I meant use `searchParams` to set the `selector` param, then serialize the entire URL object.
Also, it's not good practice to remove hashes from the URL, they serve a purpose.

Reply:
What do you exactly mean by saying `I meant use searchParams to set the selector param`. Do you mean something like this?https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/set. Or do you mean that when i set `url.searchParams` like `url.searchParams = selector=${value}` the url will change?

Reply:
> What do you exactly mean by saying `I meant use searchParams to set the selector param`. Do you mean something like this? https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/set. Or do you mean that when i set `url.searchParams` like `url.searchParams = selector=${value}` the url will change?

Okay, i get it now :D thanks!